Common/cpp/SharedItems/FrozenObject.cpp

```diff
@@ -8,7 +8,11 @@ FrozenObject::FrozenObject(jsi::Runtime &rt, const jsi::Object &object, NativeRe
   auto propertyNames = object.getPropertyNames(rt);
   for (size_t i = 0, count = propertyNames.size(rt); i < count; i++) {
     auto propertyName = propertyNames.getValueAtIndex(rt, i).asString(rt);
-    map[propertyName.utf8(rt)] = ShareableValue::adapt(rt, object.getProperty(rt, propertyName), module);
+    std::string nameStr = propertyName.utf8(rt);
+    map[nameStr] = ShareableValue::adapt(rt, object.getProperty(rt, propertyName), module);
+    if (map[nameStr]->type == ValueType::HostFunctionType) {
```

Comment:
It's not enough. You have to check if a property contains HostFunction. Imagine the following case: object->object->hostFunction.

Reply:
It also has to be done for worklets and frozenArrays.

Reply:
> Imagine the following case: object->object->hostFunction.
If you mean that a host function is in the nested object then that nested object would be checked here. Is there any place other than [this](https://github.com/software-mansion/react-native-reanimated/blob/master/Common/cpp/SharedItems/ShareableValue.cpp#L96) where `this->hostFunction` is being assigned?
> It also has to be done for worklets and frozenArrays.
Why? What case would it cover?(not mentioning that frozen arrays aren't really frozen)

Reply:
Like I said it's not enough because the function can have a reference to the other object.

Reply:
added checks 668fe6e